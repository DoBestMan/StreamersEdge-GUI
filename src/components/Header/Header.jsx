import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {NavLink} from 'react-router-dom';
import {AppActions, ModalActions} from '../../actions';
import {GenUtil} from '../../utility';
import {ModalTypes, RouteConstants as Routes} from '../../constants';
import HeaderLogo from '../../assets/images/se-logo.png';
import loginIcon from '../../assets/images/loginicon.png';
import loginIconActive from '../../assets/images/loginicon_active.png';
import menuIcon from '../../assets/images/menuicon.png';
const translate = GenUtil.translate;
class Header extends Component {
  openLoginModal = () => {
    this.props.setModalType(ModalTypes.LOGIN);
    this.props.toggleModal();
  };

  render() {
    let logButton = (
      <span onClick={ this.openLoginModal } className='header__link'>
        {translate('header.login')}
      </span>
    );

    if (this.props.isLoggedIn) {
      logButton = (
        <span onClick={ this.props.logout } className='header__link'>
          {translate('header.logout')}
        </span>
      );
    }

    return (
      <>
        <div className='header'>
          <div className='header--left'>
            <img src={ menuIcon } alt='menuIcon' />
            <NavLink exact className='header__link' activeClassName='' to=''>
              {translate('header.menu')}
            </NavLink>
            <NavLink exact className='header__link' activeClassName='' to=''>
              {translate('header.popular')}
            </NavLink>
          </div>
          <div className='header--center'>
            <img className='header__image' src={ HeaderLogo } alt='Header' />
          </div>
          <div className='header--right'>
            <NavLink exact className='header__link' activeClassName='header__link--active' to={ Routes.SIGN_UP }>
              {translate('header.signup')}
            </NavLink>
            {logButton}
            <img src={ this.props.path === '/login' ? loginIconActive : loginIcon } alt='avatar' />
          </div>
        </div>
        <div className='divider-radial' />
      </>
    );
  }
}

const mapStateToProps = (state) => ({
  isModalOpen: state.getIn(['modal', 'isOpen']),
  modalType: state.getIn(['modal', 'type']),
  isLoggedIn: state.getIn(['profiles', 'isLoggedIn'])
});

const mapDispatchToProps = (dispatch) => bindActionCreators(
  {
    toggleModal: ModalActions.toggleModal,
    setModalType: ModalActions.setModalType,
    logout: AppActions.logout
  },
  dispatch
);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
