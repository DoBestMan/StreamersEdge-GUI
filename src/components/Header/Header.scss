@import '../../assets/styles/common';

.header {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  position: relative;
  width: 100vw;
  height: $height__header;
  background-image: linear-gradient($blue-darkest, $blue);

  &__image {
    max-width: 100%;
    height: auto;
    max-height: 108.19px;
  }

  &__link {
    margin-right: 12px;
    font-size: 28px;
    color: $blue-light;
    text-decoration: none;
    cursor: pointer;

    &__active, &:hover {
      color: $white;
    }
  }

  &__logo {
    margin: auto;
    cursor: pointer;

    &--display-none {
      display: none;
    }
  }

  &--left {
    display: flex;
    align-items: center;
    margin-right: auto;
    margin-left: 50px;
  }

  &--center {
    display: flex;
    flex: 0 1 auto;
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
  }

  &--right {
    display: flex;
    align-items: center;
    margin-right: 50px;
    margin-left: auto;
  }
}
