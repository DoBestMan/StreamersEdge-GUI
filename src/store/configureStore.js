import {applyMiddleware, createStore} from 'redux';
import {createHashHistory} from 'history';
import {routerMiddleware} from 'connected-react-router';
import Immutable from 'immutable';
import {composeWithDevTools} from 'redux-devtools-extension';
import createRootReducer from '../reducers/index';
import thunk from 'redux-thunk';

// Session history
export const history = createHashHistory();

export default function configureStore() {
  // For typescript, this may need tweaking...
  let initialState = Immutable.Map();

  // Define var for filtering actions when debugging redux to reduce clutter and increase performance.
  let actionsWhitelist = [];

  // Configure enhancer for redux dev tools extensions (if available)
  const composeEnhancers = composeWithDevTools({
    features: {
      dispatch: true
    },
    // Option for immutable
    serialize: {immutable: Immutable},
    actionsWhitelist: actionsWhitelist
  });

  // Construct enhancer
  const enhancer = composeEnhancers(
    applyMiddleware(thunk, routerMiddleware(history))
  );
  const store = createStore(
    createRootReducer(history),
    initialState,
    enhancer
  );

  return store;
}