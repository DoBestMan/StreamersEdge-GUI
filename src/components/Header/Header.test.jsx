import React from 'react';
import {shallow} from 'enzyme';
import {Provider} from 'react-redux';
import configureStore from '../../store/configureStore';
import Header from './Header';

describe('<Home/>', () => {
  const store = configureStore();
  test('should render correctly', () => {
    const component = shallow(
      <Provider store={ store }>
        <Header />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
