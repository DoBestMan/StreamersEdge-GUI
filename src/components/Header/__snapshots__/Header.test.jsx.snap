// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> should render correctly 1`] = `ShallowWrapper {}`;
